### PR TITLE
Audit log service support

### PR DESCRIPTION
**Pull request by PavelLoparev**

### CODE REVIEW

File: examples/audit-log-example.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    $response = false;
+    $auditLog = \Smartling\AuditLog\AuditLogApi::create($authProvider, $projectId);
+    $st = microtime(true);
+
+    try {
+        $searchParams = (new \Smartling\AuditLog\Params\SearchRecordParameters())
+            ->setSearchQuery('foo:bar')
+            ->setOffset(0)
+            ->setLimit(100)
+            ->setSort('_id', \Smartling\AuditLog\Params\SearchRecordParameters::ORDER_DESC);
```
Comment (dimitrystd) [inline examples/audit-log-example.php]: Let's do example more real life and use `time` field.

File: src/AuditLog/AuditLogApi.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        $instance = new self($projectId, $client, $logger, self::ENDPOINT_URL);
+        $instance->setAuth($authProvider);
+
+        return $instance;
+    }
+
+    public function createProjectLevelLogRecord(CreateRecordRecommendedParameters $createRecordRecommendedParameters) {
+        $requestData = $this->getDefaultRequestData('json', $createRecordRecommendedParameters->exportToArray());
+        $endpoint = vsprintf('projects/%s/logs', [
```
Comment (dimitrystd) [inline src/AuditLog/AuditLogApi.php]: Propose move these strings to consts.

File: src/AuditLog/Params/CreateRecordRecommendedParameters.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Smartling\AuditLog\Params;
+
+use InvalidArgumentException;
+use Smartling\Parameters\BaseParameters;
+
+class CreateRecordRecommendedParameters extends BaseParameters
```
Comment (dimitrystd) [inline src/AuditLog/Params/CreateRecordRecommendedParameters.php]: `CreateRecordParameters`

**Review by dimitrystd — CHANGES_REQUESTED** [review]

File: examples/audit-log-example.php (hunk 1 above)
Comment (PavelLoparev) [reply examples/audit-log-example.php]: You mean to use `time` in `setSearchQuery()` method?

File: src/AuditLog/AuditLogApi.php (hunk 2 above)
Comment (PavelLoparev) [reply src/AuditLog/AuditLogApi.php]: Only for this class or for all others in this repo?

File: src/AuditLog/Params/CreateRecordRecommendedParameters.php (hunk 3 above)
Comment (PavelLoparev) [reply src/AuditLog/Params/CreateRecordRecommendedParameters.php]: Agree.

File: src/AuditLog/Params/CreateRecordRecommendedParameters.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+{
+    const ACTION_TYPE_UPLOAD = 'UPLOAD';
+    const ACTION_TYPE_DOWNLOAD = 'DOWNLOAD';
+    const ACTION_TYPE_CANCEL = 'CANCEL';
+    const ACTION_TYPE_DELETE = 'DELETE';
+    const ACTION_TYPE_LOCK_FIELDS = 'LOCK_FIELDS';
+    const ACTION_TYPE_UPDATE_SETTINGS = 'UPDATE_SETTINGS';
```
Comment (PavelLoparev) [inline src/AuditLog/Params/CreateRecordRecommendedParameters.php]: @dimitrystd are you ok with these action types? I've just figured out that user can manually upload context and run cron (`UPLOAD_CONTEXT` and `RUN_CRON` or `RUN_[cron_queue]_JOB`).

File: src/AuditLog/AuditLogApi.php (hunk 2 above)
Comment (dimitrystd) [reply src/AuditLog/AuditLogApi.php]: Let's start with this class only

File: examples/audit-log-example.php (hunk 1 above)
Comment (dimitrystd) [reply examples/audit-log-example.php]: I mean give people the right example. In most cases, sort by time is only one reasonable sort option.

Comment (PavelLoparev) [reply examples/audit-log-example.php]: Done

File: src/AuditLog/AuditLogApi.php (hunk 2 above)
Comment (PavelLoparev) [reply src/AuditLog/AuditLogApi.php]: Done

File: src/AuditLog/Params/CreateRecordRecommendedParameters.php (hunk 3 above)
Comment (PavelLoparev) [reply src/AuditLog/Params/CreateRecordRecommendedParameters.php]: Done

**Pull request merged**